packages/server-wallet/src/onchain-service/store.ts

```diff
@@ -0,0 +1,183 @@
+import {Bytes32} from '@statechannels/client-api-schema';
+import {providers} from 'ethers';
+
+import {
+  ChannelEventRecord,
+  OnchainServiceStoreInterface,
+  ContractEventName,
+  ChannelEventRecordMap,
+  MinimalTransaction,
+  TransactionStatus,
+  TransactionStatusEventMap,
+  TransactionStatuses,
+  NoncedMinimalTransaction,
+  TransactionSubmissionStoreInterface,
+} from './types';
+
+// TODO: This will definitely change when we use to a more permanent store
+type TransactionModel<T extends TransactionStatus> = {
+  status: T;
+  data: TransactionStatusEventMap[T];
+};
+
+export class TransactionSubmissionStore implements TransactionSubmissionStoreInterface {
+  // NOTE: V0 has an in-memory store, which will eventually be
+  // shifted to more permanent storage
+  private readonly transactions: Map<
+    string,
+    TransactionModel<keyof typeof TransactionStatuses>[]
+  > = new Map();
+
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  constructor() {}
+
+  // Used to save a transaction *before* it is sent to mempool
+  saveTransactionRequest(channelId: Bytes32, tx: NoncedMinimalTransaction): Promise<void> {
+    const existing = this.transactions.get(channelId) || [];
+    const idx = existing.findIndex(
+      e => e.status === 'pending' && (e.data as NoncedMinimalTransaction).nonce === tx.nonce
+    );
+    return Promise.resolve(this._updateTransactions(channelId, idx, {status: 'pending', data: tx}));
```

Comment:
I don't think it's necessary to wrap return values in `Promise.resolve` really. Why not use the `async` function modifier and `await`?

Reply:
It's not async since it's all just mapping functions.

I just used `Promise.resolve` instead of typing the function as `async` when there's nothing to `await` yet. I figure may as well do this, since in v1 the calls to storage will definitely be `async`